scripts/import/import_oldhgmd.pl

```diff
@@ -0,0 +1,306 @@
+#!/usr/bin/env perl
+use strict;
+use DBI;
+use Socket;
+use Bio::EnsEMBL::Registry;
+use Getopt::Long;
+use POSIX qw(strftime);
+use Cwd qw(cwd);
+
+$| = 1;
+socketpair(CHILD, PARENT, AF_UNIX, SOCK_STREAM, PF_UNSPEC) or die "ERROR: Failed to open socketpair: $!";
+PARENT->autoflush(1);
+
+my $registry = 'Bio::EnsEMBL::Registry';
+my $config = configure();
+
+my $reg_file = $config->{registry}; # the reg_file needs to be added to the config.
+$registry->load_all($reg_file); # loading the registry is important 
+my $db_adaptor = $registry->get_DBAdaptor("homo_sapiens", "variation");
+my $dbh = $db_adaptor->dbc;
+my $dbname = $dbh->dbname;
+debug($config, "Connected to the new database $dbname");
+
+my $old_dbh;
+my $old_reg_file = $config->{old_registry} if (defined ($config->{old_registry}));
+my $release =  $config->{release} if (defined ($config->{release}));
+
+my $old_var_file = "old_variation.txt";
+my $sorted_old_var = "old_new_variation.txt";
+my $new_var = "new_variation.txt";
+my $old_var_far_file = "old_variation_feature.txt";
+my $sorted_old_var_feat = "old_new_variation_feature.txt";
+my $new_var_feat = "new_variation_feature.txt";
+my $old_var_set_file = "old_variation_set.txt";
+my $sorted_old_var_set = "old_new_variation_set.txt";
+my $new_var_set = "new_variation_set.txt";
+
+
+if ($old_reg_file) {
+  $registry->load_all($old_reg_file);
+  my $db_adaptor = $registry->get_DBAdaptor("homo_sapiens", "variation");
+  # Connect to DBI
+  $old_dbh = $db_adaptor->dbc;
+  my $old_dbname = $old_dbh->dbname;
+  debug($config, "Connected to the old database $old_dbname");
+} else {
+  my $old_release = $release - 1;
+  my $old_dbname = $dbh->dbname =~ s/_${release}_/_${old_release}_/gr;
+  my $old_host = $dbh->host;
+  my $old_port = $dbh->port;
+  #Connect to DBI
+  $old_dbh = DBI->connect("DBI:mysql:database=${old_dbname};host=${old_host};port=${old_port}", "ensro", "");
+  debug($config, "Connected to the old database $old_dbname");
+}
+
+my $TMP_DIR = $config->{tmp};
+
+debug($config, "Dumping old variation tables with HGMD as source from old database");
+dump_vdata_into_file($old_dbh);
+
+debug($config, "Sorting files based on the variation_id column");
```

Comment:
The variation feature file should be sorted based on seq_region_id, seq_region_start, seq_region_end.

Reply:
You can sort the file again after you created the new file with the new ids.